src/app/qgsrastercalcdialog.cpp

```diff
@@ -95,6 +95,15 @@ QgsRasterCalcDialog::QgsRasterCalcDialog( QgsRasterLayer *rasterLayer, QWidget *
   mOutputLayer->setDialogTitle( tr( "Enter result file" ) );
   mOutputLayer->setDefaultRoot( settings.value( QStringLiteral( "/RasterCalculator/lastOutputDir" ), QDir::homePath() ).toString() );
   connect( mOutputLayer, &QgsFileWidget::fileChanged, this, [ = ]() { setAcceptButtonState(); } );
+
+  connect( mUseVirtualProviderCheckBox, &QCheckBox::clicked, this, &QgsRasterCalcDialog::setOutputToVirtual );
+
+  if ( ! useVirtualProvider() )
+  {
+    mVirtualLayerLabel->hide();
+    mVirtualLayerName->hide();
+  }
```

Comment:
not possible to call setOutputToVirtual() here?

Reply:
yes it is. thank you :)